### DISCUSSION TITLE
Question about using Rollup's plugin API for file watchers in plugin

### DISCUSSION THREAD

**Discussion by leemove** [posva/unplugin-vue-router#156] [Ideas]
I noticed that  plugin has implemented a file watcher to detect changes in files and update the plugin code. I was wondering, have you used Rollup's built-in plugin API, specifically this.addWatchFile, to watch files.

这个插件自己实现了一个文件的观察者，是不是使用rollup提供的api会更合适。例如this.addWatchFile之类

**Comment by posva** [comment]
It has to work with webpack and others too. Last time I tried that API, it just didn't work at all

**Comment by leemove** [comment]
https://github.com/unjs/unplugin
![image](https://user-images.githubusercontent.com/11146494/233984653-6e272557-a855-47a0-90fe-cf32b41fd6b0.png)

**Comment by posva** [comment]
The current implementation of watchers seems good. I will change it if it needs improving
